pennylane/transforms/add_noise.py

```diff
@@ -0,0 +1,165 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
```

Comment:
The problem here is that when you do `phi = np.random.rand()`, you are already assigning the `phi` the random value and that's what goes inside the function call. So when defining the `NoiseModel` dictionary this turns out to be static, and so I don't think this would be possible to accommodate for. 

An alternative to do this would be -
```py
def noise_fn(op, **kwargs):
    qml.RX(phi=np.random.rand(), wires=op.wires)
```